src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -181,8 +181,8 @@
     public LanguageServerWrapper wrapper;
     private Project project;
     private TextDocumentIdentifier identifier;
-    private EditorMouseListener mouseListener;
-    private EditorMouseMotionListener mouseMotionListener;
+    protected EditorMouseListener mouseListener;
+    protected EditorMouseMotionListener mouseMotionListener;
```

Comment:
easy to expanding a subclass

Reply:
not to say that this change is wrong, but are you actually planning on overwriting or accessing these properties? 

If we let anyone extend our classes (which, sure, is technically already possible) we expose ourselves to the [Fragile Base Class Problem](https://de.wikipedia.org/wiki/Fragile_Base_Class_Problem). So i would rather leave these Properties private.

You can already inject custom listeners via the constructor. Is this not sufficient for your use case?

Reply:
As you said `inject custom listeners via the constructor`,  this cann't be implemented. Because these listener is fixed, cann't be changed.
![image](https://user-images.githubusercontent.com/25758750/109443223-f929e700-7a74-11eb-8674-7edd3f60a981.png)
----
Yeah, I want to extend the `EditorEventManager` class, and do some change in method `registerListeners`. So I change the fields accessibility as protected

Reply:
> not to say that this change is wrong, but are you actually planning on overwriting or accessing these properties?
> 
> If we let anyone extend our classes (which, sure, is technically already possible) we expose ourselves to the [Fragile Base Class Problem](https://en.wikipedia.org/wiki/Fragile_base_class). So i would rather leave these Properties private.
> 
> You can already inject custom listeners via the constructor. Is this not sufficient for your use case?

Taking your advice, and keep these fields protected